domains/rpc/composables/useTokenRpc.ts

```diff
@@ -0,0 +1,234 @@
+  LSP4DigitalAssetMetadataJSON,
+} from '@/types/asset'
+
+export function useTokenRpc() {
+  return (_token?: MaybeRef<Asset | null | undefined>) => {
+    const connectedProfile = useProfile().connectedProfile()
+    const profileAddress = computed(() => connectedProfile.value?.address)
+    const { selectedChainId: chainId } = useAppStore()
+
+    const queries = computed(() => {
+      const token: Asset | null = isRef(_token)
+        ? _token.value || null
+        : _token || null
+      const { address, tokenId, tokenDataURL, tokenIdFormat } = token || {}
+      const queries: QFQueryOptions[] & { token: Asset | null } = (
+        address
+          ? [
+              queryCallContract({
+                // 0
+                chainId,
+                address,
+                method: 'owner()',
+              }),
+              queryGetData({
+                // 1
+                chainId,
+                address,
+                keyName: 'LSP4Creators[]',
+              }),
+              queryCallContract({
+                // 2
+                chainId,
+                address,
+                method: 'decimals()',
+              }),
+              queryCallContract({
+                // 3
+                chainId,
+                address,
+                method: 'tokenIdsOf(address)',
+                args: [profileAddress.value || address],
+                staleTime: 250,
+              }),
+              tokenId && tokenIdFormat === 2
+                ? queryGetData({
+                    // 4
+                    chainId,
+                    address: decodeParameter(
+                      'address',
+                      tokenId
+                    ).toLowerCase() as Address,
+                    keyName: 'LSP8ReferenceContract',
+                  })
+                : queryNull(),
+              tokenId && tokenIdFormat === 2
+                ? queryGetData({
+                    // 5
+                    chainId,
+                    address: decodeParameter(
+                      'address',
+                      tokenId
+                    ).toLowerCase() as Address,
+                    keyName: 'LSP4Creators[]',
+                  })
+                : queryNull(),
+              queryGetData({
+                // 6
+                chainId,
+                address,
+                keyName: 'LSP4Metadata',
+                process: data => browserProcessMetadata(data, keccak256),
+                aggregateLimit: 1,
+                priority: Priorities.Low,
+              }),
+              tokenId
+                ? queryGetData({
+                    // 7
+                    chainId,
+                    address,
+                    tokenId,
+                    keyName: 'LSP4Metadata',
+                    process: data => browserProcessMetadata(data, keccak256),
+                    aggregateLimit: 1,
+                    priority: Priorities.Low,
+                  })
+                : queryNull(),
```

Comment:
Looks wrong that you have two queries for `LSP4Metadata`.
Shouldn't it be:
```suggestion
              tokenId
                ? queryGetData({
                    // 7
                    chainId,
                    address,
                    tokenId,
                    keyName: 'LSP4Metadata',
                    process: data => browserProcessMetadata(data, keccak256),
                    aggregateLimit: 1,
                    priority: Priorities.Low,
                  })
                : queryGetData({
                    // 6
                    chainId,
                    address,
                    keyName: 'LSP4Metadata',
                    process: data => browserProcessMetadata(data, keccak256),
                    aggregateLimit: 1,
                    priority: Priorities.Low,
                  }),
```

Reply:
Or should be actually enough to use if `tokenId` either has a value or is `undefined`:
```
              queryGetData({
                // 6
                chainId,
                address,
                tokenId,
                keyName: 'LSP4Metadata',
                process: data => browserProcessMetadata(data, keccak256),
                aggregateLimit: 1,
                priority: Priorities.Low,
              })
```

Reply:
The one is for getDataForTokenId and the other for getData. We need to get both to later on determine which one to use.

Reply:
But if `tokenId` is undefined, null or empty (or whatever is interpreted as `false`) `queryFunctions` will use `getData`. Otherwise, if defined it will use `getDataForTokenId`. The we just use the result.

This is from `queryFunctions.ts`:
<img width="398" alt="Screenshot 2024-08-09 at 12 05 09" src="https://github.com/user-attachments/assets/3b3ac71f-3bf0-4bc0-b530-673b7558eebe">

And question is:
<img width="1117" alt="Screenshot 2024-08-09 at 12 08 53" src="https://github.com/user-attachments/assets/d1443779-404f-4e52-8c4c-da1e77aad3f3">